docs: document agent-sdk-ts architecture and capabilities (#89)
enyst/OpenHands-Tab: docs/agent-sdk-architecture.md

# OpenHands Agent SDK (TypeScript) - Architecture Guide

## Overview

> The `@openhands/agent-sdk-ts` package is a complete TypeScript implementation for building AI agents with the OpenHands platform. This document provides a comprehensive architectural overview and implementation guide for the SDK.

Author: ```suggestion
The `@openhands/agent-sdk-ts` package is a complete TypeScript implementation for building AI agents with OpenHands on VSCode. This document provides an architectural overview and implementation guide for the SDK.
```

## Design Philosophy

The SDK follows these core principles:

1. **Layered Architecture** - Clear separation between runtime, LLM, tools, workspace, and protocol types
2. **Streaming-First** - All LLM interactions use streaming for real-time responsiveness
3. **Type Safety** - Complete TypeScript types with runtime validation
4. **Modularity** - Each layer can be used independently or composed together
5. **Extensibility** - Easy to add new LLM providers, tools, and workspace implementations

> 6. **Testing** - Comprehensive test coverage with deterministic fixtures

Author: ```suggestion
6. **Testing** - Tests with deterministic fixtures
```

│  │         Runtime Layer (Orchestration)            │    │
│  │  AgentOrchestrator │ EventLog │ State │ Locks   │    │
│  └──────────────┬──────────────────────────────────┘    │
│                 │                                         │
│  ┌──────────────┴──────────────┬─────────────────┐      │
│  │      LLM Layer              │   Tools Layer    │      │
│  │  Anthropic │ OpenAI-compat  │  Terminal │ File │      │
│  │  Factory │ Streaming        │  Browser │ Tasks│      │
│  └─────────────────────────────┴─────────────────┘      │
│                 │                                         │
│  ┌──────────────┴──────────────┬─────────────────┐      │
│  │    Workspace Layer          │  Protocol Types  │      │
│  │  LocalWorkspace             │  Message │ Event │      │
│  └─────────────────────────────┴─────────────────┘      │
│                                                           │
└───────────────────────────────────────────────────────────┘
```

## Layer 1: Runtime (Orchestration)

The runtime layer coordinates agent execution, manages conversation state, and orchestrates LLM interactions.

### AgentOrchestrator

> **Purpose**: Manages the complete LLM interaction lifecycle with streaming support.

Author: ```suggestion
**Purpose**: Manages the Conversation using the LLMClient with streaming support.
```

const eventLog = new EventLog();
const state = new ConversationState(eventLog);
const orchestrator = new AgentOrchestrator(client, { events: eventLog, state });

// Tools
const workspace = new LocalWorkspace('/workspace');
const terminal = new TerminalTool(workspace);
const fileEditor = new FileEditorTool(workspace);

// Execute agent
const response = await orchestrator.runChat({
  systemPrompt: 'You are a helpful assistant.',
  messages: conversation.messages,
  tools: [terminal.definition, fileEditor.definition]
});

// Handle tool calls
if (response.message.tool_calls) {
  for (const toolCall of response.message.tool_calls) {
    const args = JSON.parse(toolCall.function.arguments);
    let result;

    if (toolCall.function.name === 'terminal') {
      result = await terminal.execute(args);
    } else if (toolCall.function.name === 'file_editor') {
      result = await fileEditor.execute(args);
    }

    // Add observation event
    eventLog.addEvent({
      type: 'ObservationEvent',
      source: 'environment',
      observation: result,
      tool_name: toolCall.function.name,
      tool_call_id: toolCall.id,
      action_id: response.message.id
    });
  }

> }

Author: /gemini wdyt now?
